src/app/static/src/components/parameters/Phases.vue

```diff
@@ -74,7 +80,7 @@ export default defineComponent({
         const formatPercent = (value: number) => `${value.toFixed(2)}%`;
 
         const phaseBlocks: Ref<PhaseBlock[]> = computed(() => {
-            const firstStartDate = displayPhases.value[0].startDate;
+            const firstStartDate = displayPhases.value[0]?.startDate;
```

Comment:
If I was a nitpicker I would say that this is a bit dangerous - it relies on `daysBetween` handling an `undefined` end date. Not even sure that even gets past the type checker.

Anyway, I guess you could immediately return `[]` if `displayPhases` is empty, but it is a bit ugly, considering that the `map` ultimately does the right thing regardless.

Reply:
Yeah, fair enough, I think it should deal with the empty case separately just to be clear - clunky but explicit!